apps/prairielearn/assets/scripts/lib/questionsTable.js

```diff
@@ -224,4 +224,9 @@ onDocumentReady(() => {
       event.preventDefault();
     }
   });
+
+  const searchInputs = document.querySelectorAll('.search-input');
+  searchInputs.forEach((searchInput) => {
+    searchInput.setAttribute('aria-label', searchInput.closest('th').dataset.field);
+  });
```

Comment:
I could not find a way to add a label directly using the `bootstrap-table` API, so I added it using client side JS. Let me know if there is a better way of achieving this.

Reply:
Yeah, I poked at the `bootstrap-table` source code a bit and I don't think there's any better way to do this. Manual DOM manipulation is the way to go.

A few notes:

- The `.search-input` query selector should be scoped to the table so that it doesn't impact anything else that might be on the page. One option would be to scope the selector to `#questionsTable .search-input`).
- This code here will only run once when the table is first initialized, but `bootstrap-table` will sometimes re-render its entire DOM, e.g. when visibility for a column is turned on or off. When that happens, the attribute you added here is lost. You should instead run this in the [`onResetView` handler](https://bootstrap-table.com/docs/api/events/#onresetview), which ensures it's re-executed when the table is re-rendered. You can see an example of this here: https://github.com/PrairieLearn/PrairieLearn/blob/12fdc12b2d5469cd31042f528308943c234c6897/apps/prairielearn/assets/scripts/instructorGradebookClient.ts#L74
- The name "QID" might not be maximally useful. Consider using text like "Filter by QID".
- This code handles `<input>` element, but not `<select>` elements like the tags/topics/etc. columns use.

Reply:
In [b2ddc80](https://github.com/PrairieLearn/PrairieLearn/pull/10514/commits/b2ddc80c9af2e30a4fa6ea94027150ce2ac8b3ba), updated to include `<select>` elements as well as narrow the query selector to just the `#questionsTable`. Also, added this to an `onResetView` handler and updated the name to 'Filter by ...`.

Note, because we are now grabbing other columns as well, there are some that have titles with underscores in the name. I added `.replace()` to update these to spaces which the screen reader handles better.

Reply:
Let's continue the conversation here: https://github.com/PrairieLearn/PrairieLearn/pull/10514#discussion_r1737517389